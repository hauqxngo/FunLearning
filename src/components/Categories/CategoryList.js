import React, { useEffect, useState } from "react";
import FunLearningApi from "../../api/api";
import SearchForm from "../Forms/SearchForm";
import CategoryCard from "./CategoryCard";
import Loading from "../Navigation/Loading";
import Footer from "../Pages/Footer";

/** Shows the list of all categories from API
 *
 * Search box will filter categories to those matching
 *
 * Routed as /categories
 *
 * Routes -> {SearchForm, CategoryCard}
 */

const CategoryList = () => {
  const [categories, setCategories] = useState(null);

  useEffect(function getCategoriesOnMount() {
    search();
  }, []);

  async function search(name) {
    let categories = await FunLearningApi.getCategories(name);
    setCategories(categories);
  }

  if (!categories) return <Loading />;

  return (
    <div>
      <h3 className="text-center text-success mt-4">
        What do you want to learn today?
      </h3>
      <SearchForm searchFor={search} />
      {categories.length ? (
        <div>
          {categories.map((c) => (
            <CategoryCard
              key={c.handle}
              handle={c.handle}
              name={c.name}
              description={c.description}
            />
          ))}
        </div>
      ) : (
        <p className="mx-5">Sorry, no categories were found.</p>
      )}
      <Footer />
    </div>
  );
};

export default CategoryList;
