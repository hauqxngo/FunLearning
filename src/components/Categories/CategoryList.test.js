import React from "react";
import { render } from "@testing-library/react";
import CategoryList from "./CategoryList";

it("matches snapshot", () => {
  const { asFragment } = render(<CategoryList />);
  expect(asFragment()).toMatchSnapshot();
});
