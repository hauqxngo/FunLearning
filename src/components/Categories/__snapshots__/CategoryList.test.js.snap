// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="d-flex justify-content-center text-success my-4"
    >
      Loading...
    </h3>
    <div
      class="d-flex justify-content-center"
    >
      <div
        class="spinner-border text-success my-2"
        role="status"
      />
    </div>
  </div>
</DocumentFragment>
`;
